[Server] Allow pre-built instance handlers in element registration (#375)
modelcontextprotocol/php-sdk: CHANGELOG.md

All notable changes to `mcp/sdk` will be documented in this file.

> 0.6.1

Reviewer: let's go with `0.7.0` here - it's rather a feature to me:

```suggestion
0.7.0
```

-----

> * Allow registering an element handler as a pre-built object instance (`[$instance, 'methodName']`) via `Builder::addTool()`, `addResource()`, `addResourceTemplate()`, and `addPrompt()`. `HandlerResolver` previously rejected instances with "Invalid array handler format" even though the `Handler` type already permitted them — this unblocks handler objects with constructor dependencies that the container-less fallback cannot build.

Reviewer 2: ```suggestion
* Allow `[$instance, 'methodName']` as an element handler in `Builder::addTool()`, `addResource()`, `addResourceTemplate()`, and `addPrompt()`. Unblocks handlers with constructor dependencies that the container-less `new $className()` fallback cannot build.
```
